src/Microsoft.AspNet.Server.Kestrel.Https/HttpsConnectionFilter.cs

```diff
@@ -3,29 +3,39 @@
 
 using System;
 using System.Net.Security;
+using System.Security.Authentication;
 using System.Security.Cryptography.X509Certificates;
 using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.AspNet.Http.Features.Internal;
 using Microsoft.AspNet.Server.Kestrel.Filter;
 
 namespace Microsoft.AspNet.Server.Kestrel.Https
 {
     public class HttpsConnectionFilter : IConnectionFilter
     {
-        private readonly X509Certificate2 _cert;
+        private readonly X509Certificate2 _serverCert;
+        private readonly ClientCertificateMode _clientCertMode;
+        private readonly ClientCertificateValidationCallback _clientValidationCallback;
         private readonly IConnectionFilter _previous;
+        private readonly SslProtocols _sslProtocols;
+        private X509Certificate2 _clientCert;
```

Comment:
HttpsConnectionFilter is per server, not per connection. We need some way to flow per connection state from OnConnection to PrepareRequest